zend_abstract_interface/zai_sapi/php5/zai_sapi.c

```diff
@@ -0,0 +1,220 @@
+#include "../zai_sapi.h"
+
+#include <main/SAPI.h>
+#include <main/php_main.h>
+#include <main/php_variables.h>
+
+#include "../zai_sapi_ini.h"
+#include "../zai_sapi_io.h"
+
+#define DEFAULT_INI          \
+    "html_errors=0\n"        \
+    "implicit_flush=1\n"     \
+    "output_buffering=0\n"   \
+    "max_execution_time=0\n" \
+    "max_input_time=-1\n"    \
+    "memory_limit=-1\n"      \
```

Comment:
Are these very lenient limits inherited from the embed sapi? I do not have a strong opinions, but at the same time making test execution unbound might not be a good practice. Do you think it make sense to start from the per-version default value and then adapt/disable this param when appropriate?

Reply:
> Are these very lenient limits inherited from the embed sapi?

Yes. These are the original embed SAPI values.

> making test execution unbound might not be a good practice. Do you think it make sense to start from the per-version default value and then adapt/disable this param when appropriate?

Yes, good catch. It would be better for unexpected behavior to be detected from the tests using the default INI values. One exception is setting `output_buffering=0`. Due to the console-ish nature of the ZAI SAPI, the PHP output buffer might get in the way of some of the tests (i.e. a test examines the output before PHP flushes the buffer). That being said, we can always explicitly enable OB for a test before MINIT with:

```c
zai_sapi_append_system_ini_entry("output_buffering", "1");
```

Reply:
Addressed in 0338206.